oauth2/oauth2-resource-server/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/resourceserver/OAuth2ResourceServerConfigurer.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2002-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.config.annotation.web.configurers.oauth2.resourceserver;
+
+import org.springframework.security.config.annotation.web.HttpSecurityBuilder;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.oauth2.core.AbstractOAuth2Token;
+import org.springframework.security.oauth2.core.OAuth2TokenVerifier;
+import org.springframework.security.oauth2.jwt.Jwt;
+import org.springframework.security.oauth2.resourceserver.web.BearerTokenResolver;
+
+/**
+ * Example configuration:
+ *
+ * oauth2().resourceServer().accessToken()
+ *     .formats()
+ *         .jwt().and()
+ *     .verifiers()
+ *         .signature().keys("http://jwk.url")
+ *
+ * Or:
+ *
+ * oauth2().resourceServer().accessToken()
+ *     .formats()
+ *         .jwt()
+ *             .processor(auth0AccessTokenProcessor())
+ *             .and()
+ *         .opaque()
+ *             .processor(auth0AccessTokenProcessor())
+ *             .and()
+ *         .and()
+ *     .verifiers()
+ *          .addVerifier(claims -> {
+ *              if ( claims.get("iss") == null ) {
+ *                  throw new OAuth2AuthenticationException(...);
+ *              }
+ *          })
+ *
+ * @author Josh Cummings
+ */
+public class OAuth2ResourceServerConfigurer<B extends HttpSecurityBuilder<B>> extends
+		AbstractHttpConfigurer<OAuth2ResourceServerConfigurer<B>, B> {
+
+	private AccessTokenFormatsConfigurer accessTokenFormatsConfigurer;
+	private AccessTokenVerifiersConfigurer accessTokenVerifiersConfigurer;
+
+	public OAuth2ResourceServerConfigurer<B> bearerTokenResolver(BearerTokenResolver resolver) {
+		return this;
+	}
+
+	public AccessTokenConfigurer accessToken(OAuth2TokenVerifier... verifiers) {
+		return new AccessTokenConfigurer();
+	}
+
+	public class AccessTokenConfigurer {
+		public AccessTokenVerifiersConfigurer verifiers() {
+			return new AccessTokenVerifiersConfigurer();
+		}
+
+		public AccessTokenFormatsConfigurer formats() {
```

Comment:
What is the purpose of `formats()`? I understand it's directly associated to the format of the access token. However, doesn't `accessToken().jwt()` imply the JWT access token format?

Reply:
To be honest, I like the symmetry. Here are the access token formats I support: --formats listed--, here are the access token verifiers I support: --verifiers listed--.

Rob makes a similar comment as you, and I'm happy to defer, especially because it more likely that both your suggestions will be in line with what has been done in the past.

I don't have any functional reason, I just think it reads better.

Reply:
See earlier comment